### PR TITLE
*: move RegionMeta to raftstore

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

Issue Number: Ref #12842 

This PR is extracted from #13334.
What's Changed:
```commit-message
This PR moves `RegionMeta` to raftstore module, so that both v1 and v2
can use it for debugging. Several fields are re-arranged to be more
concise and more informations.
```

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility
  The response of HTTP API /region/id is changed to include more fields. Since the API is used mainly for debugging, it may breaks some OP scripts.

### Release note 

```release-note
Change the response of peer query to include more information.
```

### CODE REVIEW

**Comment by BusyJay** [comment]
PTAL

File: components/raftstore/src/store/region_meta.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
```
Comment (Connor1996) [inline components/raftstore/src/store/region_meta.rs:1]: 2020 -> 2022

**Review by Connor1996 — APPROVED** [review]
the title should be `Introduce RegionMeta for debugging`?

**Comment by BusyJay** [comment]
No, `RegionMeta` is defined already in server module. I just move it to raftstore.

**Comment by Connor1996** [comment]
> No, `RegionMeta` is defined already in server module. I just move it to raftstore.

Seems you didn't remove the origin

**Review by 5kbpers — APPROVED** [review]

File: components/raftstore-v2/src/router/response_channel.rs [hunk 2]
```diff
@@ -201,10 +213,66 @@ impl<'a, Res> Future for WaitResult<'a, Res> {
...
+    pub async fn result(mut self) -> Option<Res> {
+        WaitResult { core: &self.core }.await
+    }
+}
+
+unsafe impl<Res: Send> Send for BaseSubscriber<Res> {}
+unsafe impl<Res: Send> Sync for BaseSubscriber<Res> {}
+
+/// A base channel that contains most common implementation of channels.
+pub struct BaseChannel<Res> {
```
Comment (Connor1996) [inline components/raftstore-v2/src/router/response_channel.rs:233]: miss the review of the previous PR, why use `ManuallyDrop`

**Comment by BusyJay** [comment]
/hold

It was added to control the drop order. But now there is only one field, it's not necessary anymore.

**Pull request merged**